embedded-graphics/src/geometry/point.rs

```diff
@@ -0,0 +1,368 @@
+///
+/// // Create a coord using the `new` constructor method
+/// let c1 = Point::new(10, 20);
+///
+/// // Create a coord using the handy `point` function
+/// let c2 = point(10, 20);
+///
+/// assert_eq!(c1, c2);
+/// ```
+///
+/// ## Create a `Point` from a Nalgebra `Vector2`
+///
+/// _Be sure to enable the `nalgebra_support` feature to get [Nalgebra] integration._
+///
+/// ```rust
+/// # #[cfg(feature = "nalgebra_support")] {
+/// use embedded_graphics::geometry::Point;
+/// use nalgebra::Vector2;
+///
+/// let n_coord = Vector2::new(10i32, 20);
+///
+/// assert_eq!(Point::from(n_coord), Point::new(10, 20));
+/// # }
+/// ```
+///
+/// ## Convert a `Vector2<u8>` into a `Point`
+///
+/// _Be sure to enable the `nalgebra_support` feature to get [Nalgebra] integration._
+///
+/// Smaller unsigned types that can be converted to `i32` are also supported in conversions.
+///
+/// ```rust
+/// # #[cfg(feature = "nalgebra_support")] {
+/// use embedded_graphics::geometry::Point;
+/// use nalgebra::Vector2;
+
+/// let n_coord = Vector2::new(10u8, 20);
+///
+/// assert_eq!(Point::from(n_coord), Point::new(10, 20));
+/// # }
+/// ```
+///
+/// [`Size`]: struct.Size.html
+/// [`Rectangle`]: ../primitives/rectangle/struct.Rectangle.html
+/// [`Vector2<N>`]: https://docs.rs/nalgebra/0.18.0/nalgebra/base/type.Vector2.html
+/// [`Vector2`]: https://docs.rs/nalgebra/0.18.0/nalgebra/base/type.Vector2.html
+/// [Nalgebra]: https://docs.rs/nalgebra
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct Point {
+    /// The x coordinate.
+    pub x: i32,
+
+    /// The y coordinate.
+    pub y: i32,
+}
+
+impl Point {
+    /// Create a new coordinate with X and Y values
+    pub const fn new(x: i32, y: i32) -> Self {
+        Point { x, y }
+    }
+
+    /// Creates a point with x and y equal to zero.
+    pub const fn origin() -> Self {
+        Point { x: 0, y: 0 }
+    }
+
+    /// Clamp coordinate components to positive integer range
+    ///
+    /// ```
+    /// # use embedded_graphics::geometry::Point;
+    /// #
+    /// let point = Point::new(-5, 10);
+    ///
+    /// assert_eq!(point.clamp_positive(), Point::new(0, 10));
+    /// ```
+    pub fn clamp_positive(&self) -> Self {
+        Point::new(self.x.max(0), self.y.max(0))
+    }
+
+    /// Remove the sign from a coordinate
+    ///
+    ///
+    /// ```
+    /// # use embedded_graphics::geometry::Point;
+    /// #
+    /// let point = Point::new(-5, -10);
+    ///
+    /// assert_eq!(point.abs(), Point::new(5, 10));
+    /// ```
+    pub fn abs(&self) -> Self {
+        Point::new(self.x.abs(), self.y.abs())
+    }
+}
+
+impl Add for Point {
+    type Output = Point;
+
+    fn add(self, other: Point) -> Point {
+        Point::new(self.x + other.x, self.y + other.y)
+    }
+}
+
+impl Add<Size> for Point {
+    type Output = Point;
+
+    fn add(self, other: Size) -> Point {
+        Point::new(self.x + other.width as i32, self.y + other.height as i32)
```

Comment:
As mentioned in some of my previous PRs, what if `u32::max_value()` is used here? Would it be worth changing `Size` to use `u16`s for storage so this conversion is supported? Embedded targets _probably_ don't need more than 65k values but someone out there might disagree.

There are some other cases down this file that would also suffer from the same problem.

Reply:
I was expecting this comment because we did talk about this before.

We could change the conversion to use `try_into` and panic if the conversion from `u32` to `i32` fails.

Reply:
We could, although I don't like the idea of panicking over a simple `+`. Do you know of any other `impl`s of `Add` out in the wild that panic in cases like this?

Reply:
The normal primitive integer types panic on overflow in debug builds. Should we change this to be checked in debug builds and panic but just overflow in release builds?

Reply:
Sounds good to me. If we can stick as close to Rust's own behaviour as possible that would be best. Can you add some doc comments for this and other places that panic/overflow?

Reply:
I've added some comments and debug assertions to check if width or height are too large.